pbj-integration-tests/build.gradle.kts

```diff
@@ -71,9 +72,34 @@ testing {
                 maxHeapSize = "4096m"
             }
         }
+
+        tasks.register<Test>("integTest") {
+            testClassesDirs = sources.output.classesDirs
+            classpath = sources.runtimeClasspath
+
+            shouldRunAfter(tasks.test)
+
+            useJUnitPlatform { excludeTags("FUZZ_TEST") }
+            enableAssertions = true
+        }
+
+        tasks.register<Test>("fuzzTest") {
+            testClassesDirs = sources.output.classesDirs
+            classpath = sources.runtimeClasspath
+
+            shouldRunAfter(tasks.test)
+
+            useJUnitPlatform { includeTags("FUZZ_TEST") }
+            enableAssertions = false
+        }
     }
 }
 
+tasks.test.configure() {
+    actions.clear()
```

Comment:
This is a bit of a nasty hack. 🤔 Have you considered using this task – `test` – for the "normal" integration tests and not add an additional `integTest` task? Because the tests are in `test` and usually it is expected that the task `test` executes the tests in that directory. I think this also leads to unexpected behavior when executing a single test from IDEA (if you care about that).

You could also consider to:
- Move all tests into a folder called `integTest` and then register a new test suite for that folder.
- (Maybe) Move the FuzzTests into a separate folder called `fuzzTest` and register a test suite for that. Then you do not need the filtering by tag.

Registering the test suites automatically sets up the tasks with the corresponding names. The Gradle configuration would look like this:

```
testing {
    suites { 
        register<JvmTestSuite>("integTest") { 
            dependencies {
                implementation(project()) 
            }
            targets.all {
                tasks.test { dependsOn(testTask) }
            }
        }
        register<JvmTestSuite>("fuzzTest") { 
            dependencies {
                implementation(project()) 
            }
            targets.all {
                testTask { enableAssertions = false }
                tasks.test { dependsOn(testTask) }
            }
        }
    }
}
```

Reply:
Thanks for the suggestion! That indeed looks cleaner. I'll try to implement this and see how it goes.

Reply:
Unfortunately, I didn't manage to get this working. For one, all the jvm args had to be repeated in each suite definition separately, even though the `enableAssertions` is the only difference between the two. I also had to extend the configurations for the new targets from ordinary targets similar to 

https://docs.gradle.org/current/userguide/java_testing.html#sec:configuring_java_integration_tests

otherwise Gradle wouldn't run them.

But apart from these minor things, ultimately I was unable to make the Gradle script compile the generated source at `pbj-integration-tests/build/generated/source/pbj-proto/test/` and then include the compiled classes into the classpath of the `fuzzTest` target. I'm sure I missed something trivial, but due to my poor expertise in Gradle unfortunately I was unable to find a solution for this.

While I agree that the solution in the current fix where we redefine the `test` target doesn't look conventional, it's pretty straightforward from the Gradle script perspective and gets the job done as far as running both the regular integ and the new fuzz tests is concerned. However, if you know how to solve the issues I described above, I'd be happy to learn.

Reply:
Thanks for giving it a try. If the `generated/source/pbj-proto/test/` makes this complicated (I didn't think much about that) then I agree that we should keep things in `src/test`.

My remark on keeping the setup this way is that I would then keep using `test` to run the normal integration tests. Which means **not** do `actions.clear()` and not do an additional `tasks.register<Test>("integTest")`

If you keep things as they are now, you could remove the whole

```
    targets.all {
            testTask {
                // We are running a lot of tests 10s of thousands, so they need to run in parallel
                systemProperties["junit.jupiter.execution.parallel.enabled"] = true
                systemProperties["junit.jupiter.execution.parallel.mode.default"] = "concurrent"
                // us parallel GC to keep up with high temporary garbage creation, and allow GC to use 40% of CPU if needed
                jvmArgs("-XX:+UseParallelGC", "-XX:GCTimeRatio=90")
//                jvmArgs("-XX:+UseZGC","-XX:ZAllocationSpikeTolerance=2")
//                jvmArgs("-XX:+UseG1GC", "-XX:GCTimeRatio=90", "-XX:MaxGCPauseMillis=100")
                // Some also need more memory
                minHeapSize = "512m"
                maxHeapSize = "4096m"
            }
        }

```

block. Because `targets.all { testTask {} }` is the same as `test.tasks {}`. The `all` might suggest that you configure more than one task, but it is not the case. It only configures the task associated with the test suite, and there is always only one – https://docs.gradle.org/current/userguide/jvm_test_suite_plugin.html#test_suite_target. The ability to add a new task to a test suite is still a missing feature. I know this part is quite confusing. ☹️ 

What you probably want to do is (if the `jvmArgs ` and `memory` settings are needed? Seems to work without as well 😄):

```
tasks.withType<Test>().configureEach {
   // ...
}
```

Reply:
Thanks for the tip! In the latest commit, I'm only adding the `fuzzTest` target and making the `test` depend on it, and it all seems to work fine, and it looks a lot nicer in the Gradle script itself. Please have a look.

Reply:
Perfect. Gradle part looks good to me now!